File: litho-rendercore/src/main/java/com/facebook/rendercore/annotations/Action.java
// (c) Facebook, Inc. and its affiliates. Confidential and proprietary.

package com.facebook.rendercore.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * To be used with a RenderUnit: annotates that this method is an action that can be taken on this
 * RenderUnit from the UI thread.
 */
@Target({ElementType.METHOD})
@Retention(RetentionPolicy.SOURCE)
public @interface Action {}

File: litho-rendercore/src/main/java/com/facebook/rendercore/annotations/Action.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.facebook.rendercore.annotations

/**
 * To be used with a RenderUnit: annotates that this method is an action that can be taken on this
 * RenderUnit from the UI thread.
 */
@Target(
    AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.SOURCE)
annotation class Action
